test/targets/create-target.e2e-spec.ts

```diff
@@ -77,8 +78,21 @@ describe('POST /targets', () => {
       })
 
       it('should return the newly created target', async () => {
-        const target = await targets.last()
-        expect(response.body).toEqual(TargetDto.from(target))
+        firstTarget = await targets.last()
+        expect(response.body.target).toEqual(TargetDto.from(firstTarget))
+      })
+      
+      describe('when there is another target nearby with the same topic', () => {
+        it('should create a new match', async () => {
+          response = await request(app.getHttpServer())
+            .post('/targets')
+            .set('Authorization', `Bearer ${accessToken}`)
+            .send({ ...mockTarget2, topicId: mockTopic.id })
+            .expect('Content-Type', /json/)
+            .expect(201)
+          firstTarget.topic = undefined
+          expect(response.body.matches[0]).toEqual(TargetDto.from(firstTarget))
```

Comment:
This seems like it belongs in another test

Reply:
It's something that happens on target creation... I don't see why it should be somewhere else. I have most of what is needed to test this feature on this test, why would I duplicate all of that? I'm asking because I really don't know, this is my first time writing tests in a project.

Reply:
Tests should only test one thing, that helps to stop them from growing and getting out of control. Some people tend to translate this to having just one assert per test block, but it's a bit more conceptual. For example, I see this as two tests one is checking that the request has the expected type and another that the response has the expected type. You could also see it as testing the creation as a whole, which includes the correct response code and the correct response body